server_ext_test.go

```diff
@@ -97,3 +103,85 @@ func (s) TestServer_MaxHandlers(t *testing.T) {
 		t.Fatal("Received unexpected RPC error:", err)
 	}
 }
+
+// Tests the case where the stream worker goroutine option is enabled, and a
+// number of RPCs are initiated around the same time that Stop() is called. This
+// used to result in a write to a closed channel. This test verifies that there
+// is no panic.
+func (s) TestStreamWorkers_RPCsAndStop(t *testing.T) {
+	ss := stubserver.StartTestService(t, nil, grpc.NumStreamWorkers(uint32(runtime.NumCPU())))
+	// This deferred stop takes care of stopping the server when one of the
+	// below grpc.Dials fail, and the test exits early.
+	defer ss.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	const numChannels = 20
+	const numRPCLoops = 20
+
+	// Create a bunch of clientconns and ensure that they are READY by making an
+	// RPC on them.
+	ccs := make([]*grpc.ClientConn, numChannels)
+	for i := 0; i < numChannels; i++ {
+		var err error
+		ccs[i], err = grpc.Dial(ss.Address, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		if err != nil {
+			t.Fatalf("[iteration: %d] grpc.Dial(%s) failed: %v", i, ss.Address, err)
+		}
+		defer ccs[i].Close()
+		client := testgrpc.NewTestServiceClient(ccs[i])
+		if _, err := client.EmptyCall(ctx, &testpb.Empty{}, grpc.WaitForReady(true)); err != nil {
```

Comment:
Nit: delete this `WaitForReady` as it should not be necessary and could persist a mistaken understanding that it might be necessary.  (We used to have a bug waaay back when that would have required it.)

Reply:
Done.